src/resources/cpp/lambdas.md

```diff
@@ -0,0 +1,244 @@
+library can often have their behavior changed using something callable. Creating a whole function is often excessive,
+and we use lambdas instead:
+
+`int non_a_amount = std::ranges::count_if("abcaabbac", [](char c) { return c != 'a'; });`
+
+This counts all results where the lambda returns true, in this case if the latter is not 'a'.
+
+Another example is where we want to print each value in an array:
+
+```cpp
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+int main()
+{
+    std::vector<int> v{3, -4, 2, -8, 15, 267};
+
+    // Here we declare a local lambda object, called print
+    auto print = [](const int& n) { std::cout << n << '\n'; };
+
+    // Here we use this lambda object in the algorithm for each, where the lambda will be called for each
+    // iteration and print the value in the vector.
+    std::for_each(v.cbegin(), v.cend(), print);
+
+    // As a lambda is similar to a function we can also call it like a function. This will print 5
+    print(5);
+
+    return 0;
+}
+```
+
+## Structure of lambdas
+
+A lambda consists of at least three parts:
+
+- `[]`, the introducer, containing captures.
+- `()`, the parameters (optional in some cases).
+- `{}`, the body of the lambda.
+
+There are many further optional parts, but `[](){}` or `[]{}` are the bare minimum, depending on the language version.
+
+for a lambda the return type does not need to be specified like functions do. The return type is deduced by the
+compiler. However you can specify a trailing return type to the lambda.
+
+```cpp
+// This lambda return an interger value
+[]() -> int {
+    return 5;
+};
+
+// ``equivalent to the following lambda
+[](){
+    return 5;
+};
+
+```
+
+### Parameters `()`
+
+The parameter part of a lambda is the exact same as the parameter list of a function. Here you specify the inputs of
+your lambda.
+
+For example `[](int a, int b){}` is a lambda that takes in 2 integer values.
+
+### body `{}`
+
+The lambda body is the exact same as the body of the function. It's hold the code that the lambda will execute when it's
+invoked.
+
+### Capture list
+
+As the name implies, the capture list allows us to capture values into the lambda as it's constructed. Unlike the
+parameters where we specify which values we want to pass in when we invoke the lambda. This tends to be particularly
+common when you want to bring external state into a generic algorithm.
+
+Take for example a simple program where you want to multiply every number with a number the user provides.
+
+```cpp
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+int main()
+{
+    std::vector<int> v{3, -4, 2, -8, 15, 267};
+
+    std::cout << "Please enter a number to multiply with";
+    int multiply_number = 0;
+    std::cin >> multiply_number;
+
+    // Here we declare a local lambda object. We capture the variable multiply_number so it can be used in the lambda. We return the value of the passed in value with multiply_number.
```

Comment:
Consider: "local" feels like an extra word
Consider: "We capture" -> "The lambda captures" (and a corresponding change to the rest of the sentence"
Consider: "We return" -> "The lambda returns"